compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9130,9 +9130,9 @@ private STNode parseOnFailCheck() {
         STNode failKeyword = parseFailKeyword();
         STNode varName = parseVariableName();
         STNode rightDoubleArrow = parseDoubleRightArrow(ParserRuleContext.RIGHT_DOUBLE_ARROW);
-        STNode errConstructor = parseErrorConstructorExpr(false);
+        STNode expr = parseExpression(OperatorPrecedence.EXPRESSION_ACTION, true, false);
```

Comment:
should this be `OperatorPrecedence.REMOTE_CALL_ACTION`?

Reply:
Shall we update the doc comment for the new grammar also?

Reply:
Updated [8a51481](https://github.com/ballerina-platform/ballerina-lang/pull/41098/commits/8a514812a9aa20e6f470e046c943e55c47d782f5)